packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/create-virtual-cluster.ts

```diff
@@ -0,0 +1,153 @@
+   */
+  static fromCluster(cluster: eks.ICluster): EksClusterInput {
+    return new EksClusterInput(cluster.clusterName);
+  }
+
+  /**
+   * Use a Task Input for the cluster name.
+   *
+   * @param taskInput Task Input object that accepts multiple types of payloads, in this case a literal string.
+   * @returns The value of a Task Input object, or a literal string.
+   */
+  static fromTaskInput(taskInput: sfn.TaskInput): EksClusterInput {
+    return new EksClusterInput(taskInput.value);
+  }
+
+  /**
+   * Initializes the clusterName
+   *
+   * @param clusterName The name of the EKS Cluster
+   */
+  private constructor(readonly clusterName: string) { }
+}
+
+/**
+ * Properties to define a EMR Containers CreateVirtualCluster Task on an EKS cluster
+ */
+export interface EmrContainersEksCreateVirtualClusterProps extends sfn.TaskStateBaseProps {
+
+  /**
+   * EKS Cluster or task input that contains the name of the cluster
+   */
+  readonly eksCluster: EksClusterInput;
+
+  /**
+   * The namespace of an EKS cluster
+   *
+   * @default - 'default'
+   */
+  readonly eksNamespace?: string;
+
+  /**
+   * Name of the virtual cluster that will be created.
+   * @default - the id of the state machine execution that runs this task
+   */
+  readonly virtualClusterName?: string;
+
+  /**
+   * The tags assigned to the virtual cluster
+   *
+   * @default {}
+   */
+  readonly tags?: { [key: string]: string };
+}
+
+/**
+ * Task that creates an EMR Containers virtual cluster from an EKS cluster
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-emr-eks.html
+ */
+export class EmrContainersEksCreateVirtualCluster extends sfn.TaskStateBase {
+
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: EmrContainersEksCreateVirtualClusterProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+    validatePatternSupported(this.integrationPattern, EmrContainersEksCreateVirtualCluster.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this.taskPolicies = this.createPolicyStatements();
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('emr-containers', 'createVirtualCluster', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject({
+        Name: this.props.virtualClusterName ?? sfn.JsonPath.stringAt('$$.Execution.Id'),
+        ContainerProvider: {
+          Id: this.props.eksCluster.clusterName,
+          Info: {
+            EksInfo: {
+              Namespace: this.props.eksNamespace ?? 'default',
```

Comment:
There is already a namespace in K8s with the name `default` - [Link](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/) used for objects with no other namespace. Is the intention here to use this default namespace (and avoid explicit creation) if the user hasn't specified any? I would suggest explicitly calling this out in the documentation. Also create a constant to hold the 'default' value.

Reply:
Exactly, 'default' will be use if no namespace is provided. This is documented [here](https://github.com/aws/aws-cdk/pull/15262/files/674da28c60620ece78894ec376b1bc3c58bb9e64#diff-47be2dc4953dbb968ca576c2500858e86157099e8f10017e62443952339fc0ecR68)

Reply:
Also, I see no need for a variable here. The pattern of `expression ?? default` is widely used in TypeScript